oop/oop.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
+
+# :nodoc:
+class Station
+  attr_reader :trains
+
+  def initialize(name)
+    @name = name
+    @trains = []
+  end
+
+  def host(train)
+    @trains << train if train.current == self
+  end
+
+  def depart(train)
+    @trains.delete(train) if train.current == self
+  end
+
+  def list_specififc(type)
+    @trains.select { |train| train.type == type }
+  end
+end
+
+# :nodoc:
+class Route
+  attr_reader :course
+
+  def initialize(departure, destination)
+    @course = [departure, destination]
```

Comment:
может просто "станции" ?
```suggestion
    @stations = [departure, destination]
```